===== Patch instructions generated by SAVER =====
[+] { Insert tmp := grub_scsi_open:scsi->blocksize at 5 (line 528, column 3) and replace the original expression at 25 (line 619, column 4) under true }
===== Actual patch in diff format =====
@@ -527,6 +527,7 @@ grub_scsi_open (const char *name, grub_disk_t disk)
 
   for (p = grub_scsi_dev_list; p; p = p->next)
     {
+      tmp = scsi->blocksize;
       if (p->open (id, bus, scsi))
 	{
 	  grub_errno = GRUB_ERR_NONE;
@@ -617,7 +618,7 @@ grub_scsi_open (const char *name, grub_disk_t disk)
 	{
 	  grub_free (scsi);
 	  return grub_error (GRUB_ERR_IO, "invalid sector size %d",
-			     scsi->blocksize);
+			     tmp);
 	}
       for (disk->log_sector_size = 0;
 	   (1U << disk->log_sector_size) < scsi->blocksize;
